tests/initial-setup.spec.ts

```diff
@@ -116,28 +168,35 @@ test('Default microphone is selected for built-in', async ({ page, context }) =>
     channels: 1,
   });
 
-  await page.getByTestId('built-in').click();
+  await test.step('Go to computers mic and select mic input', async () => {
+    await pages.inputSelectionPage.selectComputersMicrophone();
+    await expect(pages.computersMicConnectionPage.micInputNameElement).toContainText(micNameDef);
+  });
 
-  await expect(page.getByTestId('selected-mic')).toContainText('Default device');
-  // Select some different mic
-  await page.getByTestId('back-button').click();
-  await page.getByTestId('advanced').click();
-  await page.getByTestId('player-0-input').click();
-  await page.getByTestId('player-1-input').click();
+  await test.step('Go to advanced setup and toggle mic input', async () => {
+    await pages.computersMicConnectionPage.goBackToInputSelectionPage();
+    await pages.inputSelectionPage.selectAdvancedSetup();
+    await pages.advancedConnectionPage.expectMicInputNameToBe(blueMicNum, micNameDef);
 
-  // Make sure it keeps the different mic
-  await page.getByTestId('back-button').click();
-  await page.getByTestId('built-in').click();
-  await expect(page.getByTestId('selected-mic')).toContainText('Not related');
-  await page.getByTestId('selected-mic').click();
-  await expect(page.getByTestId('selected-mic')).toContainText('Default device');
+    await pages.advancedConnectionPage.toggleMicInputName(blueMicNum);
+    await pages.advancedConnectionPage.expectMicInputNameToBe(blueMicNum, micName);
+  });
 
-  // Check if the initial setup doesn't show after refresh
-  await page.getByTestId('save-button').click();
+  await test.step('Go back to computer mic and toggle mic again', async () => {
+    await pages.advancedConnectionPage.goBackToInputSelectionPage();
+    await pages.inputSelectionPage.selectComputersMicrophone();
+    await expect(pages.computersMicConnectionPage.micInputNameElement).toContainText(micName);
 
-  await expect(page.getByTestId('sing-a-song')).toBeVisible();
+    await pages.computersMicConnectionPage.toggleMicInputName();
+    await expect(pages.computersMicConnectionPage.micInputNameElement).toContainText(micNameDef);
+  });
 
-  await page.reload();
+  await test.step('Check if the initial setup does not show after reopen page', async () => {
+    await pages.computersMicConnectionPage.goToMainMenuPage();
+    await expect(pages.mainMenuPage.singSongElement).toBeVisible();
 
-  await expect(page.getByTestId('sing-a-song')).toBeVisible();
+    await page.goto('/?e2e-test');
```

Comment:
we can ignore this part of test for firefox for now, just add
```
  test.fixme(browserName === 'firefox', 'Doesnt work on Firefox');
```

You'll have to add `browserName` to the deconstructed argument of the test function, so 
```
test('Default microphone is selected for built-in', async ({ page, context }) => {
```
should become
```
test('Default microphone is selected for built-in', async ({ page, context, browserName }) => {
```